data_structures/src/superblock.rs

```diff
@@ -445,6 +620,16 @@ impl SuperBlockState {
             }
         }
 
+        let rescue_old_votes = self.rescue_votes_mempool.clear_and_remove_votes();
```

Comment:
This returns the votes from `self.rescue_votes_mempool.received_votes` but we never insert any votes there.

Currently all the votes are inserted into `self.votes_mempool.received_votes`, so we could move this add this check to the other loop, `for sbv in old_votes` something like:

    if is_rescue_valid() { insert_rescue_vote() } else if is_valid() { insert_vote() }

And remove the `for sbv in rescue_old_votes` loop. But note that we still need to call `self.rescue_votes_mempool.clear_and_remove_votes()` to clear old votes.

Reply:
> This returns the votes from `self.rescue_votes_mempool.received_votes` but we never insert any votes there.

The rescue votes are inserted using the function "insert_rescue_vote" that is called in "add_vote"

Reply:
`insert_rescue_vote` does not modify `self.rescue_votes_mempool.received_votes`

Reply:
ahhh true!!